cmd/libsnap-confine-private/device-cgroup-support.c

```diff
@@ -292,6 +305,14 @@ static int _sc_cgroup_v2_init_bpf(sc_device_cgroup *self, int flags) {
     if (own_group == NULL) {
         die("cannot obtain own group path");
     }
+    debug("process in cgroup %s", own_group);
+    if (!_sc_is_snap_cgroup(own_group)) {
```

Comment:
With my ex-maintainer hat on, I would only do this behind a feature flag. This is one of those things that is bound to make snaps explode in an environment that outside of our probing space. I would WARN on it by default but not die unless the feature flag is on.

Reply:
But if we only warn, then I guess we should at least avoid configuring the cgroup, or we risk messing up the user session.

Anyway, what exactly is your concern here? If systemd is a hard requirement for snapd, why cannot we assume that the cgroup will have been created as we expect it?

Reply:
The initial proposal would die only if SNAPPY_TESTING is set, but per discussion this was changed to die always whenever the cgroup isn't set, as we cannot effectively set up a piece of the confinement sandbox.

FWIW, from my local testing and with Fedora cloud images, a session was always set up, even if I tried su or sudo over ssh.

Reply:
also the fact that we are only dying in cgroupsv2 land makes this a bit safer IMHO and as @bboozzoo points out this is now part of the sandbox and so we _must_ die if we can't setup that part of the sandbox